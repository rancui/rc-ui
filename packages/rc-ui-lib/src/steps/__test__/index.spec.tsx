import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Steps } from '..';
import { Icon } from '../../icon';

describe('Steps', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  test('should render correctly', () => {
    const { container } = render(
      <Steps active={0}>
        <Steps.Item>B</Steps.Item>
        <Steps.Item>A</Steps.Item>
      </Steps>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should render custom icon correctly', () => {
    const { container } = render(
      <Steps
        active={1}
        activeIcon={<Icon name="success" />}
        finishIcon={<Icon name="finish" />}
        inactiveIcon={<Icon name="fail" />}
      >
        <Steps.Item>B</Steps.Item>
        <Steps.Item>A</Steps.Item>
        <Steps.Item>C</Steps.Item>
      </Steps>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should emit click-step event when step is clicked', async () => {
    const onClickStep = jest.fn();
    const { container } = render(
      <Steps active={1} onClickStep={onClickStep}>
        <Steps.Item>A</Steps.Item>
        <Steps.Item>B</Steps.Item>
        <Steps.Item>C</Steps.Item>
      </Steps>,
    );

    await fireEvent.click(container.querySelector('.rc-step'));
    expect(onClickStep).toHaveBeenCalledTimes(0);

    await fireEvent.click(container.querySelector('.rc-step__title'));
    expect(onClickStep).toHaveBeenCalledWith(0);

    await fireEvent.click(container.querySelectorAll('.rc-step__circle-container')[2]);
    expect(onClickStep).toHaveBeenCalledTimes(2);
    expect(onClickStep).toHaveBeenLastCalledWith(2);
  });

  test('should change inactive color when using inactive-color prop', () => {
    const { container } = render(
      <Steps active={0} inactiveColor="red">
        <Steps.Item>A</Steps.Item>
        <Steps.Item>B</Steps.Item>
      </Steps>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should change inactive icon when using inactive-icon prop', () => {
    const { container } = render(
      <Steps active={0} inactiveIcon="foo">
        <Steps.Item>A</Steps.Item>
        <Steps.Item>B</Steps.Item>
      </Steps>,
    );

    expect(container.querySelector('.van-icon-foo')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('should change finish icon when using finish-icon prop', () => {
    const { container } = render(
      <Steps active={1} finishIcon="foo">
        <Steps.Item>A</Steps.Item>
        <Steps.Item>B</Steps.Item>
      </Steps>,
    );

    const firstStep = container.querySelector('.rc-step');
    expect(firstStep.querySelector('.van-icon-foo')).toBeTruthy();
    expect(firstStep).toMatchSnapshot();
  });

  test('should render icon-prefix correctly', () => {
    const { container } = render(
      <Steps active={1} iconPrefix="custom-icon">
        <Steps.Item>A</Steps.Item>
        <Steps.Item>B</Steps.Item>
      </Steps>,
    );

    const steps = container.querySelector('.rc-step');
    expect(steps).toMatchSnapshot();
  });
});
