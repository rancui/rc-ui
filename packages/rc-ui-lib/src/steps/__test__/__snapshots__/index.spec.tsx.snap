// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps should change finish icon when using finish-icon prop 1`] = `
<div
  class="rc-hairline rc-step rc-step--horizontal rc-step--finish"
>
  <div
    class="rc-step__title"
  >
    A
  </div>
  <div
    class="rc-step__circle-container"
  >
    <i
      class="rc-badge__wrapper rc-step__icon rc-step__icon--finish van-icon van-icon-foo"
    />
  </div>
  <div
    class="rc-step__line"
  />
</div>
`;

exports[`Steps should change inactive color when using inactive-color prop 1`] = `
<div>
  <div
    class="rc-steps rc-steps--horizontal"
  >
    <div
      class="rc-steps__items"
    >
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--process"
      >
        <div
          class="rc-step__title rc-step__title--active"
        >
          A
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper rc-step__icon rc-step__icon--active van-icon van-icon-checked"
          />
        </div>
        <div
          class="rc-step__line"
          style="background: red;"
        />
      </div>
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--waiting"
      >
        <div
          class="rc-step__title"
          style="color: red;"
        >
          B
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-step__circle"
            style="background: red;"
          />
        </div>
        <div
          class="rc-step__line"
          style="background: red;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps should change inactive icon when using inactive-icon prop 1`] = `
<div>
  <div
    class="rc-steps rc-steps--horizontal"
  >
    <div
      class="rc-steps__items"
    >
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--process"
      >
        <div
          class="rc-step__title rc-step__title--active"
        >
          A
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper rc-step__icon rc-step__icon--active van-icon van-icon-checked"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--waiting"
      >
        <div
          class="rc-step__title"
        >
          B
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper rc-step__icon van-icon van-icon-foo"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps should render correctly 1`] = `
<div>
  <div
    class="rc-steps rc-steps--horizontal"
  >
    <div
      class="rc-steps__items"
    >
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--process"
      >
        <div
          class="rc-step__title rc-step__title--active"
        >
          B
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper rc-step__icon rc-step__icon--active van-icon van-icon-checked"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--waiting"
      >
        <div
          class="rc-step__title"
        >
          A
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-step__circle"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps should render custom icon correctly 1`] = `
<div>
  <div
    class="rc-steps rc-steps--horizontal"
  >
    <div
      class="rc-steps__items"
    >
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--finish"
      >
        <div
          class="rc-step__title"
        >
          B
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper van-icon van-icon-finish"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--process"
      >
        <div
          class="rc-step__title rc-step__title--active"
        >
          A
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper van-icon van-icon-success"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
      <div
        class="rc-hairline rc-step rc-step--horizontal rc-step--waiting"
      >
        <div
          class="rc-step__title"
        >
          C
        </div>
        <div
          class="rc-step__circle-container"
        >
          <i
            class="rc-badge__wrapper van-icon van-icon-fail"
          />
        </div>
        <div
          class="rc-step__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Steps should render icon-prefix correctly 1`] = `
<div
  class="rc-hairline rc-step rc-step--horizontal rc-step--finish"
>
  <div
    class="rc-step__title"
  >
    A
  </div>
  <div
    class="rc-step__circle-container"
  >
    <i
      class="rc-step__circle"
    />
  </div>
  <div
    class="rc-step__line"
  />
</div>
`;
